database/sqlite/sqlite_aclk.c

```diff
@@ -157,274 +253,66 @@ void sql_delete_aclk_table_list(struct aclk_database_worker_config *wc, struct a
-        wc->next = aclk_thread_head;
-        aclk_thread_head = wc;
-    }
-    uv_mutex_unlock(&aclk_async_lock);
+    debug(D_ACLK_SYNC,"Scheduling aclk sync table check for node %s", (char *) argv[0]);
+    struct aclk_database_cmd cmd;
+    memset(&cmd, 0, sizeof(cmd));
+    cmd.opcode = ACLK_DATABASE_DELETE_HOST;
+    cmd.param[0] = strdupz((char *) argv[0]);
+    aclk_database_enq_cmd_noblock(&cmd);
+    return 0;
 }
 
-void aclk_del_worker_thread(struct aclk_database_worker_config *wc)
-{
-    if (unlikely(!wc))
-        return;
-
-    uv_mutex_lock(&aclk_async_lock);
-    struct aclk_database_worker_config **tmp = &aclk_thread_head;
-    while (*tmp && (*tmp) != wc)
-        tmp = &(*tmp)->next;
-    if (*tmp)
-        *tmp = wc->next;
-    uv_mutex_unlock(&aclk_async_lock);
-}
+#define SQL_SELECT_ACLK_ACTIVE_LIST "SELECT REPLACE(SUBSTR(name,19),'_','-') FROM sqlite_schema " \
+        "WHERE name LIKE 'aclk_chart_latest_%' AND type IN ('table');"
 
-int aclk_worker_thread_exists(char *guid)
+static void sql_check_aclk_table_list(void)
 {
-    int rc = 0;
-    uv_mutex_lock(&aclk_async_lock);
-
-    struct aclk_database_worker_config *tmp = aclk_thread_head;
-
-    while (tmp && !rc) {
-        rc = strcmp(tmp->uuid_str, guid) == 0;
-        tmp = tmp->next;
+    char *err_msg = NULL;
+    debug(D_ACLK_SYNC,"Cleaning tables for nodes that do not exist");
+    int rc = sqlite3_exec_monitored(db_meta, SQL_SELECT_ACLK_ACTIVE_LIST, sql_check_aclk_table, NULL, &err_msg);
+    if (rc != SQLITE_OK) {
+        error_report("Query failed when trying to check for obsolete ACLK sync tables, %s", err_msg);
+        sqlite3_free(err_msg);
     }
-    uv_mutex_unlock(&aclk_async_lock);
-    return rc;
 }
 
-void aclk_database_init_cmd_queue(struct aclk_database_worker_config *wc)
+static int sql_maint_aclk_sync_database(void *data __maybe_unused, int argc __maybe_unused, char **argv, char **column __maybe_unused)
 {
-    wc->cmd_queue.head = wc->cmd_queue.tail = 0;
-    wc->queue_size = 0;
-    fatal_assert(0 == uv_cond_init(&wc->cmd_cond));
-    fatal_assert(0 == uv_mutex_init(&wc->cmd_mutex));
-}
-
-int aclk_database_enq_cmd_noblock(struct aclk_database_worker_config *wc, struct aclk_database_cmd *cmd)
-{
-    unsigned queue_size;
-
-    /* wait for free space in queue */
-    uv_mutex_lock(&wc->cmd_mutex);
-    if ((queue_size = wc->queue_size) == ACLK_DATABASE_CMD_Q_MAX_SIZE || wc->is_shutting_down) {
-        uv_mutex_unlock(&wc->cmd_mutex);
-        return 1;
-    }
-
-    fatal_assert(queue_size < ACLK_DATABASE_CMD_Q_MAX_SIZE);
-    /* enqueue command */
-    wc->cmd_queue.cmd_array[wc->cmd_queue.tail] = *cmd;
-    wc->cmd_queue.tail = wc->cmd_queue.tail != ACLK_DATABASE_CMD_Q_MAX_SIZE - 1 ?
-                         wc->cmd_queue.tail + 1 : 0;
-    wc->queue_size = queue_size + 1;
-    uv_mutex_unlock(&wc->cmd_mutex);
+    char sql[512];
+    snprintf(sql,511, "DELETE FROM aclk_alert_%s WHERE date_submitted IS NOT NULL AND "
```

Comment:
On line `275` (#define SQL_SELECT_ACLK_ACTIVE_LIST) and `300` (`#define SQL_SELECT_ACLK_ALERT_LIST`) we have sql queries as `define`, but here it is given direct as argument. The biggest difference is how we are passing arguments using strings (`%` in the others and `%s` and `%d` here). Are you sure we cannot move this `DELETE` for a define to keep one unique pattern in the code?

Reply:
Another change here, please, use `snprintfz` instead `snprintf`.